### DISCUSSION TITLE
[Discuss] Establish skywalking-grafana-showcase repository

### DISCUSSION THREAD

**Discussion by wu-sheng** [apache/skywalking#10378] [Ideas]
## Resolution

Based on @kezhenxu94 suggestion, we are going to keep Grafana relative showcase at https://github.com/apache/skywalking-showcase. No new repository will be created.
___

Hi team

As some of you have been tracking, we(Kai and I) are building PromQL support[1] in SkyWalking.
The purpose of creating this support is
1. Grafana users to set up UI on their own
2. Adopt/integrate PromQL-supported platforms for alerting, AIOps, visualization, etc.

Right now, we can share that, we are very optimistic about providing this in the 9.4.0 release soon as the first MVP version.
By that, I want to open this discussion about whether we should build a new showcase repository for the Grafana dashboard configurations we built on top of these PromQL APIs.

The purposes of this repository are
1. Demonstrate how to use various types of SkyWalking metrics on Grafana
2. Separate any of these setups from official releases, including OAP and UI
3. These dashboards are not SkyWalking showcases, as we will never be able to release those dashboard JSON(s). These JSONs are written by us, we own the IP, but, they require Grafana in the runtime, which is licensed in AGPL.

We need to be clear in a separate repository, this repository is only for Grafana end-user convenience. SkyWalking full stack is running out of Grafana dependencies.

## Repository Description
SkyWalking Grafana Showcase is an extension of SkyWalking showcase, to provide additional dashboard settings with demo Grafana deployments for end-user convenience.

<Notice, we would copy the skywalking-showcase in this repository for re-use>

The PromQL feature should be ready in two weeks, so, if you have any concerns about the license or any suggestions, please comment here.

[1] https://github.com/apache/skywalking/issues/10080

Official mail list link, https://lists.apache.org/thread/opfyy5k775d0co4hkyynth7hg8rl79rn

**Comment by kezhenxu94** [comment]
If we are going to copy the showcase repo, what's the reason not to add Grafana related things to that repo?

**Reply by wu-sheng** [reply]
1. Grafana is AGPL. 
2. I think we are not going to maintain all the dashboards for all layouts about this. Too many works. We just want to show something. 

Of course, due to <2>, we could put this repository out of ASF. This is an option too.

**Reply by kezhenxu94** [reply]
> 1. Grafana is AGPL. 

I doubt this is an issue. Grafana being AGPL-licensed doesn't make any difference for it to be a separate repo or part of showcase repo, does it?

**Reply by kezhenxu94** [reply]
> These JSONs are written by us, we own the IP, but, they require Grafana in the runtime, which is licensed in AGPL. 

And I also doubt this is an issue. Including our own configuration files for an AGPL-licensed software would be a license issue? That sounds really mind blowing to me. 

And since I believe it's not an issue I never plan to remove the JSON files for Grafana in our main repo, if you insist it's an issue then we should remove those in our main repo.

**Reply by wu-sheng** [reply]
License concerns are about not hosting this in the main repository, not for showcase.
The reasons of proposing are two things.
Keeping in showcase is fine from license perspective. I was thinking maybe that makes it more like we are going to provide grafana dashboard settings for all features.

**Comment by wu-sheng** [comment]
To reduce maintaining load, let's see whether others have concerns to keep this in the showcase.

**Comment by chenyi19851209** [comment]
We have integrated skywalking and grafana (loki) to do some trace and log linkage, so we pay more attention to this feature.

**Comment by wu-sheng** [comment]
Based on @kezhenxu94 suggestion, we are going to keep Grafana relative showcase at https://github.com/apache/skywalking-showcase. No new repository will be created.
